Handlers/LocalizationAttributePartHandler.cs

```diff
@@ -0,0 +1,137 @@
+                    }
+                    else {
+                        context.Metadata.DisplayText += T(" (culture undefined)").Text;
+                    }
+                }
+                if (context.ContentItem.HasDraft() && !context.ContentItem.IsPublished()) {
+                    //ContentItem is draft
+                    context.Metadata.DisplayText += T(" (draft)").Text;
+                }
+            }
+        }
+
+        protected override void UpdateEditorShape(UpdateEditorContext context) {
+            base.UpdateEditorShape(context);
+
+            //Here we implement localization logic
+            LocalizationPart locPart = context.Content.As<LocalizationPart>();
+            ProductAttributesPart attributesPart = context.ContentItem.As<ProductAttributesPart>();
+            if (locPart != null && attributesPart != null) {
+                if (attributesPart.AttributeIds.Count() > 0) {
+                    var settings = attributesPart.TypePartDefinition.Settings.GetModel<ProductAttributeLocalizationSettings>();
+                    
+                    if (settings.TryToLocalizeAttributes) {
+                        //try to replace attributes with their correct localization
+                        //newAttributesIds is IEnumerable<Tuple<int, int>>.
+                        //newAttributesIds.Item1 is the attribute id in the initial ProductAttributesPart
+                        //newAttributesIds.Item2 is the attribute id after localization
+                        var newAttributeIds = _attributeService.GetAttributes(attributesPart.AttributeIds)
+                            .Select(pap => {
+                                var ci = pap.ContentItem;
+                                if (_localizationService.GetContentCulture(ci) == locPart.Culture.Culture) {
+                                    //this attribute is fine
+                                    return new Tuple<int, int>(ci.Id, ci.Id);
+                                }
+                                var localized = _localizationService.GetLocalizations(ci)
+                                    .FirstOrDefault(lp => lp.Culture == locPart.Culture);
+                                return localized == null ?
+                                    new Tuple<int, int>(ci.Id, -ci.Id) : //negative id where we found no localization
+                                    new Tuple<int, int>(ci.Id, localized.Id);
+                            });
+
+                        if (newAttributeIds.Any(ni => ni.Item2 < 0)) {
+                            if (settings.RemoveAttributesWithoutLocalization) {
+                                //remove the items for which we could not find a localization
+                                _orchardServices.Notifier.Warning(T(
+                                    "We could not find a correct localization for the following attributes, so they were removed: {0}",
```

Comment:
If this method was called during a save/publish, and there is no error, the attributes will not be in (whatever version of) the product we end up with. Even if there is an error, the attribute will be unselected. I am changing this string to: "... they were removed from this product: {0}"